rest/route/host_spawn.go

```diff
@@ -71,6 +72,22 @@ func (hph *hostPostHandler) Run(ctx context.Context) gimlet.Responder {
 		hph.options.SetDefaultTimeZone(user.Settings.Timezone)
 	}
 
+	if !user.HasDistroCreatePermission() {
+		d, err := distro.FindOneId(ctx, hph.options.DistroID)
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding distro '%s'", hph.options.DistroID))
+		}
+		if d == nil {
+			return gimlet.MakeJSONErrorResponder(errors.Errorf("distro '%s' not found", hph.options.DistroID))
+		}
+		if d.AdminOnly {
```

Comment:
Could we comment why we need to check the distro create permission and why it not being admin only allows the operation to succeed?

Is it that when a user cannot create distros, they are not an admin so if the distro is admin only they cannot spawn it?

Reply:
Create distro is a proxy check for "is this person a distro admin" because that's the only unique permission that's tied to being a distro admin. I can clarify that in the helper function.

Also added a comment that spawning an admin-only distro is restricted to distro admins (i.e. those with distro create permission).